neos/neos-development-collection
# BUGFIX: site list in "content" of left drawer policy checked

Using policies for NodeTreePrivilege (in Policy.yaml) did not reflect in the list of sites in left drawer "content" site list. Now a restricted user only sees sites with NodeTreePrivilege access GRANTED.

Function buildSiteList is enriched by $context and $privilegeManager. On each site the node of path `/sites/{root-node-name-of-site}` is checked by `$privilegeManager->isGranted()` and put into site list on GRANTED.

**Use case / Review instructions**

- start with a fresh NEOS development installtion (see https://github.com/neos/neos-development-collection#contributing)
- kickstart multi site setup (First.Site, Second.Site, Third.Site) by command 
`$ for s in First Second Third; do ./flow kickstart:site $s.Site $s ; done`

- Import sites just created
`$ for s in First Second Third; do ./flow site:import --package-key $s.Site ; done`

- domain setup for sites in backed Administration/Sites Management
First.Site -> https://first.tds (_tds_ is my development domain)
Second.Site -> https://second.tds
Third.Site -> https://third.tds
![image](https://user-images.githubusercontent.com/19776511/205447369-7406e6b5-5bcc-48bc-8b22-5bf9fe2cd8da.png)

- Setup `Policy.yaml` (see [Policy.yaml.txt](https://github.com/neos/neos-development-collection/files/10146777/Policy.yaml.txt) ) and check resulting roles by command
```
$ ./flow security:listroles
+-----------------------------------------------+-----------------------+
| Id                                            | Label                 |
+-----------------------------------------------+-----------------------+
| Neos.Neos:LivePublisher                       | Live publisher        |
| Neos.Neos:Administrator                       | Neos Administrator    |
| Neos.Neos:RestrictedEditor                    | Restricted Editor     |
| Neos.Neos:Editor                              | Editor                |
| Neos.Neos:UserManager                         | Neos User Manager     |
| Neos.RedirectHandler.Ui:RedirectAdministrator | RedirectAdministrator |
| Neos.Setup:SetupUser                          | SetupUser             |
| First.Site:FirstEditor                        | First site Editor     |
| Second.Site:SecondEditor                      | Second site Editor    |
| Third.Site:ThirdEditor                        | Third site Editor     |
+-----------------------------------------------+-----------------------+
```
`NodeTreePrivilege` matcher in `Policy.yaml` is `'isDescendantNodeOf("/sites/first-site")'` for First.Site and so on. Retrieve the node names by command:
```
$ ./flow site:list

 Name                 Node name                 Resources package               Status
----------------------------------------------------------------------------------------
 First                first-site                First.Site                      online
 Second               second-site               Second.Site                     online
 Third                third-site                Third.Site                      online

```
- Add test user:
`$ ./flow user:create --roles Second.Site:SecondEditor TestUser testuser-password Test User`

- Log into NEOS as administrator
![image](https://user-images.githubusercontent.com/19776511/205448906-427e46af-ec1f-481f-9e06-c9b212426dc8.png)

- Log into NEOS as TestUser
![image](https://user-images.githubusercontent.com/19776511/205448973-b80e5c70-10e4-47ca-8e67-1a1d9de220ac.png)

**Checklist**
